dao/classes/AssignmentDAO.java

```diff
@@ -0,0 +1,121 @@
+package com.solvd.db.mysql.dao.classes;
+
+import com.solvd.db.mysql.dao.AbstractDAO;
+import com.solvd.db.mysql.dao.IDAO;
+import com.solvd.db.mysql.model.Assignment;
+import com.solvd.db.mysql.model.ClassTable;
+import com.solvd.db.utils.ConnectionManager;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AssignmentDAO extends AbstractDAO<Assignment> implements IDAO<Assignment> {
+    private static final Logger logger = LogManager.getLogger(StudentDAO.class);
+    public static final String insertQuery = "insert into assignments(name, due_date, score, class_id) " +
+            "values(?,?,?,?)";
+    public static final String updateQuery = "update assignments set name = ?, due_date = ?, score = ?, class_id = ? where id = ?";
+
+    @Override
+    public boolean create(Assignment assignment) {
+        try ( Connection connection = ConnectionManager.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(insertQuery, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, assignment.getName());
+            preparedStatement.setDate(2, assignment.getDueDate());
+            preparedStatement.setInt(3, assignment.getScore());
+            preparedStatement.setLong(4, assignment.getClassId().getId());
+
+            if (preparedStatement.executeUpdate()>0){
+                ResultSet resultSet = preparedStatement.getGeneratedKeys();
+                if (resultSet.next()){
+                    long generatedId = resultSet.getLong(1);
+                    logger.info("Assignment created with ID: " + generatedId);
+                }
+            } else {
+                logger.warn("Failed to create assignment.");
+                return false;
+            }
+        } catch (SQLException e) {
+            logger.error("Error while creating assignment.", e);
+        } return false;
+    }
+
+    @Override
+    public Assignment getById(long id) {
+        Assignment assignment = new Assignment();
+        try ( Connection connection = ConnectionManager.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement("select * from assignment where id = ?");
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()){
+                assignment.setId(resultSet.getLong("id"));
+                assignment.setName(resultSet.getString("name"));
+                assignment.setDueDate(resultSet.getDate("due_date"));
+                assignment.setScore(resultSet.getInt("score"));
+                long classId = resultSet.getLong("class_id");
+                ClassDAO classDAO = new ClassDAO();
```

Comment:
Since I have a field like private ClassTable classId in the AssignmentDAO class. This is the reference object of the ClassTable class which was defined in database as a foreign key.  So i found only this way to connect 2 classes and retrieve the id value of that object. If you can to tell how to do it without creating dao inside other dao it would be really appreciated. Thank you

Reply:
I haven't tried it, but i think it could work using the concept of dependency injection